### PR TITLE
Use clone for calendar duplication

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #7576

### CODE REVIEW

**Comment by trasher** [comment]
> I just checked and it does not clone time ranges not closing times.

I'm OK to add them, but were them cloned before?

**Comment by cedric-anne** [comment]
> > I just checked and it does not clone time ranges not closing times.
> 
> I'm OK to add them, but were them cloned before?

I guess they should be cloned, code was containing.
```php
         Calendar_Holiday::cloneCalendar($oldID, $newID);	
         CalendarSegment::cloneCalendar($oldID, $newID);
```

File: inc/contract.class.php [hunk 1]
```diff
@@ -47,6 +48,15 @@ class Contract extends CommonDBTM {
    protected $usenotepad               = true;
 
 
+   public function getCloneRelations() :array {
```
Comment (cedric-anne) [inline inc/contract.class.php:51]: You should drop the `post_addItem` method or you will clone twice the item.

Comment (trasher) [reply inc/contract.class.php:51]: Done, I've dropped the two related places

File: inc/contract_item.class.php [hunk 2]
```diff
@@ -696,4 +696,24 @@ static function getRelationMassiveActionsSpecificities() {
...
+   static function getItemsAssociationRequest($itemtype, $items_id) {
+      global $DB;
+
+      return $DB->request([
+         'SELECT' => 'id',
+         'FROM'   => static::getTable(),
+         'WHERE'  => [
+            static::$items_id_1  => $items_id
+         ]
+      ]);
```
Comment (cedric-anne) [inline inc/contract_item.class.php:717]:
```suggestion
      $where = [];
      if ($itemtype == 'Contract') {
         $where = [
            static::$items_id_1  => $items_id
         ];
      } else {
         $where = [
            static::$itemstype_2  => $itemtype
            static::$items_id_2  => $items_id
         ];
      }

      return $DB->request([
         'SELECT' => 'id',
         'FROM'   => static::getTable(),
         'WHERE'  => [
            static::$items_id_1  => $items_id
         ]
      ]);
```

**Pull request merged**